Letterbook.IntegrationTests/Fixtures/CollectionSubject.cs

```diff
@@ -0,0 +1,70 @@
+using System.Collections;
+using System.Reactive.Linq;
+using System.Reactive.Subjects;
+
+namespace Letterbook.IntegrationTests.Fixtures;
+
+/// <summary>
+/// An Observable Subject that also implements ICollection, so it can be used as the backing collection for
+/// OpenTelemetry in-memory exporter
+/// </summary>
+/// <remarks>Using an Observable for this is useful because the Observable behavior makes it easy to act on spans as they arrive. A regular
+/// Collection doesn't do that, and we would need a lot more arbitrary waits and timeouts in order to be confident we had gotten the spans
+/// we need.</remarks>
+/// <typeparam name="T"></typeparam>
+public class CollectionSubject<T> : SubjectBase<T>, ICollection<T>
```

Comment:
This is an Observable subject that also implements ICollection. We use it as the backing store for the in-memory spans. This is necessary because the InMemoryExporter requires an ICollection, but normal collections won't let us react to new spans as they arrive. Since this is all happening asynchronously, that's an important feature.